### DISCUSSION TITLE
Debugger breakpoints for certain classes of instructions

### DISCUSSION THREAD

**Discussion by bachimusprime** [bochs-emu/Bochs#345] [Ideas]
Hey, so I maintain another PC emulator called PCBox, and I'd like to be able to compare it against Bochs to see where my remaining CPU bugs lie in my SSE1 and SSE2 emulation, but I can't break on specifically SSE1 or SSE2 instructions whenever they occur in Bochs. This would be a very useful feature for developers of other emulators.

**Comment by stlintel** [comment]
These things could be done using instrumentation.
You define instrumentation callback BX_INSTR_BEFORE_EXECUTION and can see if you going to execute SSE/SSE2 instruction.
And do actions you want like calling for debugger break
